### PR TITLE
option to specify the "corner roundness"

### PR DESCRIPTION
**Pull request by schaumtier**
new option "corner-roundness". "0" means no corners. "12" is the default value (which draws corners of rectangle, blur and highlight round.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,6 +65,8 @@ All configuration is done either at the config file in `XDG_CONFIG_DIR/.config/s
 fullscreen = true
 # Exit directly after copy/save action
 early-exit = true
+# Draw corners of rectangles round if the value is greater than 0 (0 disabled rounded corners)
+corner_roundness = 12,
```
Comment (LordMZTE) [inline README.md]: There shouldn't be a comma here. We should also consider writing `12.0` to clarify that this is a float, not an int.

File: src/style.rs [hunk 2]
```diff
@@ -215,11 +224,12 @@ impl Size {
         }
     }
 
-    pub fn to_corner_radius(self) -> f32 {
+    pub fn to_corner_roundness(self) -> f32 {
```
Comment (LordMZTE) [inline src/style.rs:227]: Why is this function needed? It always returns the same value, regardless of `self`. At a glance, it also seems unused.

File: src/style.rs [hunk 3]
```diff
@@ -18,6 +18,9 @@ pub struct Style {
     pub fill: bool,
 }
 
+#[derive(Clone, Copy, Debug, Default)]
+pub struct Properties {}
```
Comment (LordMZTE) [inline src/style.rs:22]: Why do we need this struct?

File: src/style.rs [hunk 4]
```diff
@@ -176,6 +179,12 @@ impl FromVariant for Size {
     }
 }
 
+impl Properties {
+    pub fn corner_roundness() -> f32 {
```
Comment (LordMZTE) [inline src/style.rs:183]: This seems a little redundant. Perhaps we could directly reference `APP_CONFIG.read().corner_roundness()` where it's used.

**Review by LordMZTE — CHANGES_REQUESTED** [review]
Thanks! I left some comments on things I believe could be improved.

File: README.md (hunk 1 above)
Comment (schaumtier) [reply README.md]: removed it already

File: src/style.rs (hunk 3 above)
Comment (schaumtier) [reply src/style.rs:22]: i don't really know - just needed a place to attach the getter function... maybe its better kept somewhere else. rust is really not my programming language

File: src/style.rs (hunk 4 above)
Comment (schaumtier) [reply src/style.rs:183]: initially i thought of the need to differentiate between "Small", "Medium" and "Large" - thats why i've made a function. but of course would it be enough to have the numeric value.
if it's okay to reference APP_CONFIG directly in e.g. rectangle.rs, i can do this there.

**Review by LordMZTE — APPROVED** [review]

**Comment by gabm** [comment]
great, thank you both!

**Pull request merged**